nest/03.advanced_concepts.md

```diff
@@ -0,0 +1,70 @@
+They are not being maintained by the Raven-OS team or someone trusted by the team.
+You must decide whether to trust their maintainers and take full responsibility for any consequences of using any unofficial repository.
+
+### 3.1.3 Configuring repositories
+The repositories configuration is in the `config.toml` file of Nest.
+It would look something like this:
+
+```toml
+# Stable repository
+[repositories.stable]
+mirrors = ["https://stable.raven-os.org"]
+
+# Beta repository (uncomment to enable)
+# [repositories.beta]
+# mirrors = ["https://beta.raven-os.org"]
+
+# Nightly repository (uncomment to enable)
+# [repositories.nightly]
+# mirrors = ["https://nightly.raven-os.org"]
+```
+
+A mirror is basically a copy of all datas from the main server.
+They exist for redundancy and speed.
+Mirrors are spread all over the globe in strategic locations.
+If you live in Sweden and use a mirror from Sweden, the download speed should be faster than if you used a mirror from Australia for example.
+The point of mirrors is to choose the nearest one from your location, so you'll download packages as fast as possible.
+It's advised to have several mirrors for each repository Nest uses, because a mirror can be down or broken for a small period of time, or just be down forever.
+If Nest can't reach a mirror, it will try with the next one in the list.
+
+Only `stable` is enabled by default, so you'll have to uncomment the other repositories and their mirrors in order to use them.
+
+You can also add an unofficial repository to this list, by following the same syntax shown above in the code excerpt.
+
+## 3.2 Overview of the cache
+Nest is using a cache to store a copy of the list of all available packages for Nest on the OS.
+Nest can only install packages present in your cache.
+That's why you need to **keep it up-to-date as often as possible**, otherwise Nest won't install new versions of packages that came out on Nest's server.
+
+There are 3 different folders for the cache:
+* `/var/nest/available` contains the list of all available packages for Nest.
+The content of this folder gets updated everytime you update your local cache with Nest.
+* `/var/nest/downloaded` has all the packages that Nest has downloaded, whether they are installed or not.
+If you installed a package before, and then uninstalled it, Nest will keep its archive there, in case you change your mind and want to install that same version of the same package later.
+This allows Nest to speed up the process of installation, as it doesn't have to download the same archive again.
+* `/var/nest/installed` contains all the packages that are currently installed on the OS.
+
+## 3.3 Version requirement
+[//]: # (TODO: add link to the section 1.4)
+As explained in [the section 1.4](), the packages have a specific syntax regarding their name.
+A thing that wasn't mentionned at the time was that you can have requirement(s) on the version of the package you want to install.
```

Comment:
I don't like the current wording of the section at all, I'd prefer something like this:

---

> "Sometimes you don't need a specific version of a package, but only that it fits certain prerequisites.
> Version requirements solve this problem by allowing you to specify a minimum or maximum version.
> 
> Examples:[...]"

Followed by examples of course.

---
The problem with the current text is that it doesn't present a reason to use this and explains nothing (you have to understand from the examples). It may be a good idea and go even further by explicitly giving the version requirement syntax.

Reply:
I'm giving examples right after.It feels redundant to me to explicitly give or explain the version requirement syntax. Like should I explain all the operators you can use for the version requirement syntax?

Reply:
It may sound stupid, but "<", "<=", "==", ">=", ">" are not necessarily the symbols someone may expect (Ex: I'm not sure if it's "=" or "=="). The order is also important, it might be a good idea to talk about semver.